cmd/exchange_wrapper_issues/main.go

```diff
@@ -0,0 +1,838 @@
+	}
+	if base.API.CredentialsValidator.RequiresPEM && keys[lowerExchangeName].PEMKey == "" {
+		keys[lowerExchangeName].PEMKey = "PEM"
+	}
+	if base.API.CredentialsValidator.RequiresClientID && keys[lowerExchangeName].ClientID == "" {
+		keys[lowerExchangeName].ClientID = config.DefaultAPIClientID
+	}
+	if keys[lowerExchangeName].OTPSecret == "" {
+		keys[lowerExchangeName].OTPSecret = "-" // Ensure OTP is available for use
+	}
+
+	base.API.Credentials.Key = keys[lowerExchangeName].Key
+	base.Config.API.Credentials.Key = keys[lowerExchangeName].Key
+
+	base.API.Credentials.Secret = keys[lowerExchangeName].Secret
+	base.Config.API.Credentials.Secret = keys[lowerExchangeName].Secret
+
+	base.API.Credentials.ClientID = keys[lowerExchangeName].ClientID
+	base.Config.API.Credentials.ClientID = keys[lowerExchangeName].ClientID
+
+	if keys[lowerExchangeName].OTPSecret != "-" {
+		base.Config.API.Credentials.OTPSecret = keys[lowerExchangeName].OTPSecret
+	}
+
+	base.API.AuthenticatedSupport = true
+	base.API.AuthenticatedWebsocketSupport = true
+	base.Config.API.AuthenticatedSupport = true
+	base.Config.API.AuthenticatedWebsocketSupport = true
+
+	authenticated = base.ValidateAPICredentials()
+
+	return authenticated
+}
+
+func parseOrderSide(orderSide string) exchange.OrderSide {
+	switch orderSide {
+	case exchange.AnyOrderSide.ToString():
+		return exchange.AnyOrderSide
+	case exchange.BuyOrderSide.ToString():
+		return exchange.BuyOrderSide
+	case exchange.SellOrderSide.ToString():
+		return exchange.SellOrderSide
+	case exchange.BidOrderSide.ToString():
+		return exchange.BidOrderSide
+	case exchange.AskOrderSide.ToString():
+		return exchange.AskOrderSide
+	default:
+		log.Printf("Orderside '%v' not recognised, defaulting to BUY", orderSide)
+		return exchange.BuyOrderSide
+	}
+}
+
+func parseOrderType(orderType string) exchange.OrderType {
+	switch orderType {
+	case exchange.AnyOrderType.ToString():
+		return exchange.AnyOrderType
+	case exchange.LimitOrderType.ToString():
+		return exchange.LimitOrderType
+	case exchange.MarketOrderType.ToString():
+		return exchange.MarketOrderType
+	case exchange.ImmediateOrCancelOrderType.ToString():
+		return exchange.ImmediateOrCancelOrderType
+	case exchange.StopOrderType.ToString():
+		return exchange.StopOrderType
+	case exchange.TrailingStopOrderType.ToString():
+		return exchange.TrailingStopOrderType
+	case exchange.UnknownOrderType.ToString():
+		return exchange.UnknownOrderType
+	default:
+		log.Printf("OrderType '%v' not recognised, defaulting to LIMIT", orderTypeOverride)
+		return exchange.LimitOrderType
+	}
+}
+
+func testWrappers(e exchange.IBotExchange, base *exchange.Base, config *Config) []ExchangeAssetPairResponses {
+	var response []ExchangeAssetPairResponses
+	assetTypes := base.GetAssetTypes()
+	testOrderSide := parseOrderSide(config.OrderSubmission.OrderSide)
+	testOrderType := parseOrderType(config.OrderSubmission.OrderType)
+	if assetTypeOverride != "" {
```

Comment:
Should check to see if the asset is valid here: asset.IsValid(assetTypeOverride) or after the CLI flags are parse